crates/client/mempool/src/block_production.rs

```diff
@@ -59,14 +59,22 @@ pub enum Error {
 fn csd_to_state_diff(
     backend: &MadaraBackend,
     on_top_of: &Option<DbBlockId>,
-    csd: &CommitmentStateDiff,
+    state_map: StateMaps,
 ) -> Result<StateDiff, Error> {
+    // `StateMaps` to `CommitmentStateDiff` conversion drops declared_contracts. This causes
+    // Cairo 0 contracts to be removed from the state diff. So we keep track of them separately.
+    let mut deprecated_declared_contracts = state_map.declared_contracts.clone();
```

Comment:
This should not be cloned. Update the conversion `From<StateMap> for CommitmentStateDiff` to include `deprecated_classes` as part of `CommitmentStateDiff`. This can be done zero-copy by mutating `state_map.declared_contracts` in place as the conversion takes ownership of the `StateMap`.

Reply:
My bad, since `CommitmentStateDiff` is part of blockifier this should be implemented via an extension trait instead.